src/main/java/com/cloudbees/jenkins/support/api/CommandOutputContent.java

```diff
@@ -0,0 +1,106 @@
+package com.cloudbees.jenkins.support.api;
+
+import com.cloudbees.jenkins.support.AsyncResultCache;
+import hudson.model.Node;
+import hudson.remoting.Callable;
+import hudson.remoting.VirtualChannel;
+import jenkins.model.Jenkins;
+
+import java.io.*;
+import java.util.WeakHashMap;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * Content of a command output. You can only instantiate this content with
+ * a builder method which launch the command immediately.
+ */
+public class CommandOutputContent extends StringContent {
+
+    private static final Logger LOGGER = Logger.getLogger(CommandOutputContent.class.getName());
+
+    private CommandOutputContent(String name, String value) {
+        super(name, value);
+    }
+
+    public static class CommandLauncher implements Callable<String, RuntimeException> {
+        final String[] command;
+
+        private CommandLauncher(String... command) {
+            this.command = command;
+        }
+
+        public String call() {
+            StringWriter bos = new StringWriter();
+            PrintWriter pw = new PrintWriter(bos);
+            try {
+                dumpCommandOutput(pw, command);
+            } catch (Exception e) {
+                e.printStackTrace(pw);
+            }
+            pw.flush();
+            return bos.toString();
+        }
+
+        @edu.umd.cs.findbugs.annotations.SuppressWarnings({"DM_DEFAULT_ENCODING", "OS_OPEN_STREAM"})
+        private static void dumpCommandOutput(PrintWriter writer, String... command) throws IOException {
+            InputStream is = new ProcessBuilder(command).start().getInputStream();
+            try {
+                // this is reading from the process so platform encoding is correct
+                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(is));
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    writer.println(line);
+                }
```

Comment:
Overkill. If you were starting with a `StringWriter` and an `InputStreamReader`, there is no need to look up line boundaries. A simple `IOUtils.copyStream` would suffice.